### PR TITLE
[Bugfix] Improve consistency in getgateway.h

### PR DESCRIPTION
**Pull request by TzviPM**
## Description

* change the file name of `getgateway.c` to `getgateway.m` for consistency (*Note**: all C is valid Objective C)
* add include of `<netinet/in.h>` to `getgateway.h` as it contains definition for `in_addr_h`
  * **Note**: without this, if a build system tries to include `getgateway.h` before `<netinet/in.h>`, then `in_addr_t` will not be defined.

## Related Issues

AFAIK, no explicit bugs exist for these changes

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by TzviPM** [comment]
@miquelbeltran are you the right person to review this? I'm not sure how to request a review, but this PR should be ready for one.

The failing check for code coverage seems irrelevant in this case, since I didn't really add any new code per say.

File: packages/network_info_plus/network_info_plus/ios/Classes/getgateway.m [hunk 1]
```diff
@@ -0,0 +1,78 @@
+//
+//  getgateway.c
+//  network_info_plus
+//
+//  Created by Dimitri Dessus on 08/07/2021.
+//
+// directly cloned from https://stackoverflow.com/a/29440193/5078902
+// thanks to Alex0072005
+
```
Comment (miquelbeltran) [inline packages/network_info_plus/network_info_plus/ios/Classes/getgateway.m:9]: Can you remove the file header?

**Comment by miquelbeltran** [comment]
Thanks! Potentially is all good, but I need to build and run at least once before approving and merging.

**Comment by TzviPM** [comment]
I think this actually can't be merged as is. The getgateway.c file was copied from a stack overflow in 2015, which was released under a [CC BY-SA 3.0](https://creativecommons.org/licenses/by-sa/3.0/) license.

Changing the file requires updating the header to specify that we made changes, linking to the license, and releasing our changes under the same license. I'm not sure how the current BSD license would have to be updated to exclude this file either.

I have asked the author if they would publish a github gist with the code and specify a more permissive license.

**Comment by miquelbeltran** [comment]
Thanks for checking! I made a comment about removing the header, but then I realized that the header was in the original file, so I removed my comment. I'm fine to keep the header as it is, or add the changes as you say.

**Comment by TzviPM** [comment]
to better track this: https://stackoverflow.com/a/29440193/2679603

**Comment by TzviPM** [comment]
Tracking issue: https://github.com/fluttercommunity/plus_plugins/issues/1058. I will close this PR for now as it will not be mergeable with the current licensing issues.

**Pull request closed without merging**